### PR TITLE
Fixed date logic for list and view all

### PR DESCRIPTION
**Pull request by ChrisKraljevicD2L**
Based on Bob's feedback on [this story](https://rally1.rallydev.com/#/89963236876d/detail/userstory/148138333680?fdp=true)

### CODE REVIEW

**Comment by CodeBaboon** [comment]
@ryantmer One thing to note if this does get combined into the date-behavior is that the fall-through case is a different format between the two uses (one includes the weekday, the other does not).

**Comment by ryantmer** [comment]
Yeah, true. There's also the langterm differences, which was the reason I didn't do it when I introduced the date-behaior a couple PRs ago. i.e. I was lazy and didn't want to figure that out 😆

**Comment by ChrisKraljevicD2L** [comment]
Is this puppy good to merge or does anyone want to test some stuff?

**Comment by CodeBaboon** [comment]
I can test some things.

**Comment by CodeBaboon** [comment]
Testing Checklist
Sanity Checks
- [ ] Verify that unit tests and spec tests have been written, if appropriate
   NOTE: Feels like at least [this test](https://github.com/Brightspace/d2l-upcoming-assessments-ui/blob/master/test/components/d2l-all-assessments-list-item.js#L102) should have to change in some way. Might also be good to introduce a similar test for the not-all-assessments-list-item logic.
- [x] Verify that unit tests, spec tests, and linting are all passing
- [x] Verify that no code review comments are outstanding
- [x] ~Verify that the code is behind a feature flag~ n/a

Acceptance/Functionality
- [x] Check that the acceptance criteria of the story has been fulfilled
- [x] Check that related functionality has not been severely affected (smoke test)
- [x] (UI) Check that the functionality is the same across browsers and devices

Accessibility (UI)
- [x] ~Verify that workflows can be completed via screen reader/keyboard~ n/a
- [x] Verify that workflows can be completed when the browser is zoomed to 200%
- [x] ~Verify that there are no obvious colour contrast/combination problems~ n/a

Internationalization (UI)
- [x] Verify that lang terms are not hard coded
- [x] Verify that workflows can be completed when RTL

**Comment by ryantmer** [comment]
46b25a3 hahaha 😛

**Comment by ChrisKraljevicD2L** [comment]
@CodeBaboon Tests have been altered, may I humbly request a quick revisit of this PR?

File: test/components/d2l-assessments-list-item.js [hunk 1]
```diff
@@ -85,4 +91,18 @@ describe('<d2l-assessments-list-item>', function() {
...
 
+	describe('getDateString', function() {
+		[
+			{ date: nowish(0), dateStr: 'today', result: /^Due Today$/ },
+			{ date: nowish(1), dateStr: 'tomorrow', result: /^Due Tomorrow$/ },
+			{ date: nowish(5), dateStr: 'date within the week', result: /^Due.*(Sun|Mon|Tues|Wednes|Thurs|Fri|Satur|Sun)day.*$/ }
+		].forEach(function(testCase) {
+			it('returns correct string for ' + testCase.dateStr, function() {
+				var element = fixture('basic');
+				var relativeDateString = element._getDateString(testCase.date, 'dueDateShort', 'dueDate');
```
Comment (CodeBaboon) [inline test/components/d2l-assessments-list-item.js]: nit: naming of this variable is copypasta

**Pull request merged**